src/components/product_list.jsx

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useRef, useState } from "react";
+import { useSelector } from "react-redux";
+
+import useIntersectionObserver from "hooks/use_intersection_observer";
+
+import Item from "components/product_item";
+import { Tray } from "styles/tray";
+
+const ProductList = ({ product_list }) => {
+  const page = useRef(4);
+  const [viewList, setList] = useState([
+    ...product_list.slice(0, page.current * 4),
+  ]);
+  const target = useRef(null);
+
+  const [observe, unobserve] = useIntersectionObserver(() => {
+    setList([...product_list.slice(0, (page.current + 1) * 4)]);
+    page.current = page.current + 1;
+  });
+
+  useEffect(() => {
+    if (page.current === 4) {
+      observe(target.current);
+    }
+
+    if (page.current * 4 >= product_list.length) {
+      unobserve(target.current);
+    }
+  }, [viewList]);
```

Comment:
dependency array 체크해보세요.
eslint-plugin-react-hooks
  
```suggestion
  }, [viewList]);
```